### PR TITLE
Added URLEncode transformer

### PR DESCRIPTION
**Pull request by adi88d**
## XSOAR [Partner](https://xsoar.pan.dev/docs/partners/become-a-tech-partner)?
In case you are a technological partner:
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**.

If you are not a technological partner (i.e. PANW employee, customer or individual developer), please delete this section.

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28694

## Description
Added URLEncode transformer.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/TestPlaybooks/playbook-URLEncode-Test.yml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    task:
+      id: 595db122-9331-45bd-88a8-9059d1b5085e
+      version: -1
+      name: URLDecode
+      description: |-
+        Converts
+        https://example.com
+        into
+        https:%2F%2Fexample.com
+      scriptName: URLEncode
```
Comment (yaakovi) [inline Packs/CommonScripts/TestPlaybooks/playbook-URLEncode-Test.yml:47]: you should use it as a transformer, not as a regular script.

File: Packs/CommonScripts/Scripts/URLEncode/URLEncode.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+def main(args):
+    value = args.get('value')
+    decoded_value = unquote(value)
+    processed_value = quote(decoded_value)
+
+    eContext = {
+        'EncodedURL': processed_value
+    }
+
+    return CommandResults(readable_output=processed_value,
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/URLEncode/URLEncode.py:17]: a transformer shouldn't have context and readable output.
no need to use `CommandResults`

File: Packs/CommonScripts/TestPlaybooks/playbook-URLEncode-Test.yml [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    skipunavailable: false
+    quietmode: 0
+  "1":
+    id: "1"
+    taskid: 595db122-9331-45bd-88a8-9059d1b5085e
+    type: regular
+    task:
+      id: 595db122-9331-45bd-88a8-9059d1b5085e
+      version: -1
+      name: URLDecode
```
Comment (yaakovi) [inline Packs/CommonScripts/TestPlaybooks/playbook-URLEncode-Test.yml:41]:
```suggestion
      name: URLEncode
```

File: Packs/CommonScripts/Scripts/URLEncode/URLEncode.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import demistomock as demisto
+from CommonServerPython import *
+from urllib.parse import quote, unquote
+
+''' MAIN FUNCTION '''
+
+
+def main(args):
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/URLEncode/URLEncode.py:8]: add sanity unit-tests:
* where input is encoded
* where input isn't encoded

File: Packs/CommonScripts/Scripts/URLEncode/URLEncode.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+args:
+- default: false
+  description: The input url
+  isArray: false
+  name: value
+  required: true
+  secret: false
+comment: |-
+  Converts
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/URLEncode/URLEncode.yml:9]: improve description

File: Packs/CommonScripts/Scripts/URLEncode/URLEncode.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    https%3A//example.com
+commonfields:
+  id: URLEncode
+  version: -1
+enabled: false
+name: URLEncode
+outputs:
+- contextPath: EncodedURL
+  description: Parsed url as key/value
+  type: String
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/URLEncode/URLEncode.yml:21]: remove

File: Packs/CommonScripts/ReleaseNotes/1_2_50.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+##### New: URLEncode
+- Converts
```
Comment (yaakovi) [inline Packs/CommonScripts/ReleaseNotes/1_2_50.md:4]: update with the new description

**Review by yaakovi — CHANGES_REQUESTED** [review]

File: Packs/CommonScripts/Scripts/URLEncode/URLEncode_test.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    When
+    - call URLEncode transformer.
+    Then
+    - validate The url didnt changed since it was already encoded.
+    """
+    res = main({'value': 'https%3A//www.google.com/'})
+    assert res == 'https%3A//www.google.com/'
+
+
+def test_URLEncode_partial_encoded_input():
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/URLEncode/URLEncode_test.py:30]: nice!

**Review by yaakovi — APPROVED** [review]

**Comment by kirbles19** [comment]
@adi88d Done

**Pull request merged**